### PR TITLE
Improve support for mocking dependencies in DI

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: lib/awilixManager.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+} from 'awilix'
+import type { Resolver } from 'awilix/lib/resolvers'
 
 declare module 'awilix' {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   interface ResolverOptions<T> {
     asyncInit?: boolean | string
     asyncInitPriority?: number // lower means it gets initted earlier
-    asyncDispose?: boolean | string | ((instance: T) => Promise<unknown>)
+    asyncDispose?: boolean | string | (<U extends T>(instance: U) => Promise<unknown>)
```
Comment (kibertoad) [inline lib/awilixManager.ts:16]: Without this change it wasn't possible to pass a class that would extend the original class as a resolver override.

E. g. something like this:
```
consumerErrorResolver: asClass(FakeConsumerErrorResolver, SINGLETON_CONFIG),
```

File: lib/awilixManager.ts [hunk 2]
```diff
@@ -21,6 +29,17 @@ export type AwilixManagerConfig = {
   strictBooleanEnforced?: boolean
 }
 
+export type ResolvedDependencies<TDependencies> = {
+  [Key in keyof TDependencies]: Resolver<TDependencies[Key]>
+}
+
+export function asMockClass<T = object>(
```
Comment (kibertoad) [inline lib/awilixManager.ts:36]: This is for cases when we know that passed class does not fully implement the original class contract, and we are fine with that.

**Review by CarlosGamero — APPROVED** [review]

File: lib/awilixManager.ts (hunk 1 above)
Comment (jarrodpayne) [reply lib/awilixManager.ts:16]: Nice! I was just writing up a repro for this! https://codesandbox.io/p/devbox/awilix-ts-interface-j5sll2

Comment (kibertoad) [reply lib/awilixManager.ts:16]: wahahaha, awesome!

**Pull request merged**